zerver/lib/events.py

```diff
@@ -687,15 +687,16 @@ def name(sub: Dict[str, Any]) -> str:
             state['max_message_id'] = -1
 
         if 'raw_unread_msgs' in state:
-            remove_id = event['message_id']
-            remove_message_id_from_unread_mgs(state['raw_unread_msgs'], remove_id)
+            remove_ids = event['message_ids']
+            for remove_id in remove_ids:
```

Comment:
Loop 2

Reply:
Hmm, yeah.  I think this is something we'll very much want to fix, in a follow-up commit.

(These changes should have been one commit earlier)